compiler/testdata/zeromap.go

```diff
@@ -0,0 +1,20 @@
+package main
+
+type hasPadding struct {
+	b1 bool
+	i  int
+	b2 bool
+}
+
+func main() {
+	m := make(map[hasPadding]int)
+	var s hasPadding
+
+	for i := 0; i < 10; i++ {
+		s.b1 = i&1 == 0
+		s.i = i
+		s.b2 = i&1 == 1
+		m[s]++
+	}
+	println(len(m))
+}
```

Comment:
Suggestion: use the following code to test:

```go
func testZeroGet(m map[hasPadding]int, s hasPadding) int {
        return m[s]
}

func testZeroSet(m map[hasPadding]int, s hasPadding) {
        m[s] = 5
}
```

This makes zeromap.ll smaller and more readable while still testing the important parts (the calls to `runtime.memzero`). For example, `testZeroGet` compiles to:

```llvm
define hidden i32 @main.testZeroGet(ptr dereferenceable_or_null(40) %m, i1 %s.b1, i32 %s.i, i1 %s.b2, ptr %context) unnamed_addr #1 {
entry:
  %hashmap.key = alloca %main.hasPadding, align 8
  %hashmap.value = alloca i32, align 4
  %s = alloca %main.hasPadding, align 8
  %0 = insertvalue %main.hasPadding zeroinitializer, i1 %s.b1, 0
  %1 = insertvalue %main.hasPadding %0, i32 %s.i, 1
  %2 = insertvalue %main.hasPadding %1, i1 %s.b2, 2
  %stackalloc = alloca i8, align 1
  store %main.hasPadding zeroinitializer, ptr %s, align 8
  call void @runtime.trackPointer(ptr nonnull %s, ptr nonnull %stackalloc, ptr undef) #3
  store %main.hasPadding %2, ptr %s, align 8
  call void @llvm.lifetime.start.p0(i64 4, ptr nonnull %hashmap.value)
  call void @llvm.lifetime.start.p0(i64 12, ptr nonnull %hashmap.key)
  store %main.hasPadding %2, ptr %hashmap.key, align 8
  %3 = getelementptr inbounds i8, ptr %hashmap.key, i32 1
  call void @runtime.memzero(ptr nonnull %3, i32 3, ptr undef) #3
  %4 = getelementptr inbounds i8, ptr %hashmap.key, i32 9
  call void @runtime.memzero(ptr nonnull %4, i32 3, ptr undef) #3
  %5 = call i1 @runtime.hashmapBinaryGet(ptr %m, ptr nonnull %hashmap.key, ptr nonnull %hashmap.value, i32 4, ptr undef) #3
  call void @llvm.lifetime.end.p0(i64 12, ptr nonnull %hashmap.key)
  %6 = load i32, ptr %hashmap.value, align 4
  call void @llvm.lifetime.end.p0(i64 4, ptr nonnull %hashmap.value)
  ret i32 %6
}
```

Also, it might be worth testing an array as well (it could just be a `[2]hasPadding`).